src/components/StoreRevokeBlock.js

```diff
@@ -1,80 +1,162 @@
 import React, { Component } from "react";
+/** @jsx jsx */
+import { css, jsx } from "@emotion/core";
 import PropTypes from "prop-types";
+import { connect } from "react-redux";
 import HashColor from "./UI/HashColor";
 import HashColorInput from "./UI/HashColorInput";
-import { OrangeButton } from "./UI/Button";
+import Modal from "./UI/Modal";
 import { isValidCertificateHash } from "../components/utils";
+import {
+  verifyCertificateValidity,
+  getCertificateValidity
+} from "../reducers/api";
+import Divider from "./UI/Divider";
+import CertificateDropZone from "./CertificateDropZone";
+import { getIsVerifying } from "../reducers/application";
+import {
+  getRevokeCertificateHash,
+  updateRevokeCertificateHash
+} from "../reducers/admin";
+
+const certificateDropzone = css`
+  width: 100%;
+  text-align: center;
+`;
 
 class StoreRevokeBlock extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      certificateHash: "",
-      certificateHashIsValid: true
+      certificate: {},
+      certificateHashIsValid: true,
+      isModalVisible: false
     };
-
     this.onHashChange = this.onHashChange.bind(this);
     this.onRevokeClick = this.onRevokeClick.bind(this);
+    this.handleCertificateSelected = this.handleCertificateSelected.bind(this);
+    this.toggleModal = this.toggleModal.bind(this);
+  }
+
+  getCertificateStatus(certValidator) {
+    const validaor = this.props.revokeCertificateValidity[certValidator];
+    const status = {
+      verified: validaor ? validaor.valid : true,
+      verifying: false,
+      error: ""
+    };
+    return status;
   }
 
   onHashChange(event) {
+    const inputCertificateHash = event.target.value;
+    this.props.updateRevokeCertificateHash(inputCertificateHash);
+  }
+
+  toggleModal() {
+    const { isModalVisible } = this.state;
+    const { revokeCertificateHash } = this.props;
     this.setState({
-      certificateHash: event.target.value,
-      certificateHashIsValid: isValidCertificateHash(event.target.value)
+      certificateHashIsValid: isValidCertificateHash(revokeCertificateHash)
     });
+    if (isValidCertificateHash(revokeCertificateHash)) {
+      this.setState({
+        isModalVisible: !isModalVisible
+      });
+    }
+  }
+
+  handleCertificateSelected(payload) {
+    this.props.verifyCertificateValidity(payload);
   }
 
   onRevokeClick() {
-    const { adminAddress, storeAddress, handleCertificateRevoke } = this.props;
-    const { certificateHash } = this.state;
+    const {
+      adminAddress,
+      storeAddress,
+      handleCertificateRevoke,
+      revokeCertificateHash
+    } = this.props;
+    if (isValidCertificateHash(revokeCertificateHash)) {
+      const payload = {
+        storeAddress,
+        fromAddress: adminAddress,
+        certificateHash: revokeCertificateHash
+      };
+      handleCertificateRevoke(payload);
+      this.toggleModal();
```

Comment:
why adding a modal ? the process is already not simple (drag/drop or need to copy paste hash), wy adding more steps ?